internal/controller/isbservicerollout/isbservicerollout_progressive.go

```diff
@@ -59,58 +58,27 @@ func (r *ISBServiceRolloutReconciler) AssessUpgradingChild(ctx context.Context,
 		return apiv1.AssessmentResultSuccess, nil
 	}
 
-	// Get all Pipelines using this "upgrading" isbsvc
-	pipelines, err := r.getPipelineListForChildISBSvc(ctx, existingUpgradingChildDef.GetNamespace(), existingUpgradingChildDef.GetName())
-	if err != nil {
-		return apiv1.AssessmentResultUnknown, fmt.Errorf("Error retrieving pipelines for isbsvc %s/%s; can't make assessment for progressive: %s",
-			existingUpgradingChildDef.GetNamespace(), existingUpgradingChildDef.GetName(), err.Error())
-	}
-	if pipelines == nil {
-		numaLogger.Debugf("Can't assess isbsvc; didn't find any pipelines yet using this isbsvc")
-		return apiv1.AssessmentResultUnknown, nil
-	}
-
-	// map each PipelineRollout to its Pipeline - if we don't have a Pipeline for any of them, it is probably still being created, so we return "Unknown"
-	rolloutToPipeline := make(map[*apiv1.PipelineRollout]*unstructured.Unstructured)
+	// for each PipelineRollout, we need to check that its current Upgrading Status is for a Pipeline which is in fact using this isbsvc
+	// otherwise, it may not have yet started the upgrade process for this isbsvc
 	for _, pipelineRollout := range pipelineRollouts {
-		foundPipeline := false
-		for _, pipeline := range pipelines.Items {
-			pipelineParent, err := ctlrcommon.GetRolloutParentName(pipeline.GetName())
-			if err != nil {
-				return apiv1.AssessmentResultUnknown, err
-			}
-			if pipelineParent == pipelineRollout.Name {
-				rolloutToPipeline[&pipelineRollout] = &pipeline
-				foundPipeline = true
-				break
-			}
-		}
-		if !foundPipeline {
-			numaLogger.Debugf("Can't assess isbsvc; didn't find a Pipeline associated with PipelineRollout %s/%s using this isbsvc",
-				pipelineRollout.GetNamespace(), pipelineRollout.GetName())
+		upgradingPipelineStatus := pipelineRollout.Status.ProgressiveStatus.UpgradingPipelineStatus
+		if upgradingPipelineStatus == nil || upgradingPipelineStatus.InterStepBufferServiceName != existingUpgradingChildDef.GetName() {
+			numaLogger.WithValues("pipelinerollout", pipelineRollout.GetName()).Debug("can't assess ISBService; pipeline is not yet upgrading with this ISBService")
```

Comment:
previously we looked for the `Pipeline` itself to determine if it's using this isbsvc. This was a problem, because the Pipeline could have already been deleted. Now we use the `PipelineRollout.ProgressiveStatus.UpgradingChildStatus.InterstepBufferServiceName` to confirm that the upgrading Pipeline that we're getting an AssessmentResult from is in fact using this isbsvc